### DISCUSSION TITLE
Cannot export site template with Get-PnPSiteTemplate

### DISCUSSION THREAD

**Discussion by RSCA001** [pnp/powershell#955] [Q&A]
I am running Powershell ISE as Admin, I connect with Connect-PnPOnline -Url https://myDomain.sharepoint.com/sites/MySite -Interactive -ForceAuthentication   with a credential that has SP Admin role

Then I run Get-PnPSiteTemplate -Out C:\MyFolder\MyTemplate.xml  which starts to run & fails immediately with the error in SiteSecurity 
 
Get-PnPSiteTemplate : Attempted to perform an unauthorized operation. 

When I connect to the site with the UseWebLogin flag, the Get-PnPSiteTemplate  runs for longer,   gets past the SiteSecurity section, and fails on a list with  Get-PnPSiteTemplate : GetAccessTokenAsync() called without an ACS token generator. Specify in AuthenticationManager constructor the authentication parameters

Is this a bug, or am I doing something wrong

**Comment by sairao77** [comment]
@RSCA001 - this is expected behavior, you are supposed to be a site collection admin on the site to export it as a template, just SP Admin role doesn't cover it. Can you try after adding yourself as site collection admin

**Reply by RSCA001** [reply]
Thanks for the reply.  I have added a group I belong to as a site owner.  Is that what you mean ?   Do I need to do something else?

**Reply by sairao77** [reply]
@RSCA001 - I meant add yourself as a site admin from SharePoint Admin Center [How to add a user as a site admin](https://docs.microsoft.com/en-us/sharepoint/manage-site-collection-administrators)

I noticed adding a group that you are part of to site owners has issues, I will recommend adding yourself separately as a site owner rather through a group and give it a try

**Reply by RSCA001** [reply]
Many Thanks.  I will give that a try,  I am not sure I have access to that, so things might go quiet while I try to get things sorted. THX again

**Reply by RSCA001** [reply]
Thank you for the detailed reply.   I have added my self as both an Additional administrator and a site owner to https://myDomain.sharepoint.com/sites/MySite.   Unfortunately everything else returns the same error messages as before, depending on the parameters I added to the Connect-PnPOnline command.  So maybe I misunderstood the instructions.   Is there a distinction between site admin & site collection admin ? I have seen both terms used

**Reply by sairao77** [reply]
@RSCA001 - There is not distinction, sorry if I caused a confusion. Site Collection Admin is the right role for the above command to work.
